File: backend/user-service/src/main/java/com/mall/user/entity/Consultation.java
package com.mall.user.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("consultation")
public class Consultation {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private Long productId;
    @TableField("content")
    private String question;
    @TableField("reply_content")
    private String answer;
    private Integer status; // 0=pending, 1=replied
    private LocalDateTime createdTime;
}

File: frontend/src/client/api/consultation.ts
import request from './request'

export interface Consultation {
  id: number
  userId: number
  productId: number
  question: string
  answer: string | null
  status: number
  createdTime: string
}

export function getConsultationList() {
  return request.get<Consultation[]>('/user/consultation/list')
}

export function addConsultation(data: { productId: number; question: string }) {
  return request.post<void>('/user/consultation/add', data)
}
